pkg/pillar/cmd/zedrouter/flowstats.go

```diff
@@ -670,8 +742,106 @@ func DNSStopMonitor(bnNum int) {
 	}
 }
 
+func checkDADProbe(ctx *zedrouterContext, bnNum int, packet gopacket.Packet) {
+	var needUpdate, foundSrcMac bool
+	var vifInfo []types.VifNameMac
+	var netstatus types.NetworkInstanceStatus
+	var vifTrig types.VifIPTrig
+
+	// use the IPAssigments of the NetworkInstanceStatus, since this is switched net
+	// and the field will not be assigned or modified by others
+	pub := ctx.pubNetworkInstanceStatus
+	items := pub.GetAll()
+	for _, st := range items {
+		netstatus = st.(types.NetworkInstanceStatus)
+		if netstatus.Type != types.NetworkInstanceTypeSwitch || netstatus.BridgeNum != bnNum {
+			continue
+		}
+		vifInfo = netstatus.Vifs
+		break
+	}
+	if len(vifInfo) == 0 { // there is no Mac on the bridge
+		log.Tracef("checkDADProbe: no mac on the bridge")
+		return
+	}
+
+	var etherPkt *layers.Ethernet
+	etherLayer := packet.Layer(layers.LayerTypeEthernet)
+	if etherLayer != nil {
+		etherPkt, _ = etherLayer.(*layers.Ethernet)
+		for _, vif := range vifInfo {
+			if strings.Compare(etherPkt.SrcMAC.String(), vif.MacAddr) == 0 {
+				foundSrcMac = true
+				break
+			}
+		}
+	}
+	if !foundSrcMac {
+		log.Tracef("checkDADProbe: pkt no dst mac for us\n")
+		return
+	}
+
+	ip6Layer := packet.Layer(layers.LayerTypeIPv6)
+	if ip6Layer == nil {
+		return
+	}
+	ip6 := ip6Layer.(*layers.IPv6)
+	// We are looking for ICMPv6 Neighbor solicitation packet that
+	// tries to find if the address calculated locally is a duplicate.
+	// Such packets have a source IP of all zeroes (::)
+	if ip6.SrcIP.String() != "::" {
+		return
+	}
+
+	icmp6Layer := packet.Layer(layers.LayerTypeICMPv6NeighborSolicitation)
+	if icmp6Layer == nil {
+		return
+	}
+	for _, vif := range vifInfo {
+		if strings.Compare(vif.MacAddr, etherPkt.SrcMAC.String()) == 0 {
+			icmp6, _ := icmp6Layer.(*layers.ICMPv6NeighborSolicitation)
+			log.Tracef("ICMPv6: TargetAddress %s", icmp6.TargetAddress.String())
+
+			if _, ok := netstatus.IPAssignments[vif.MacAddr]; !ok {
+				log.Functionf("checkDADProbe: mac %v assign new IPv6 address %v\n",
+					vif.MacAddr, icmp6.TargetAddress)
+				netstatus.IPAssignments[vif.MacAddr] = []net.IP{icmp6.TargetAddress}
```

Comment:
IPAssignments is part of network instance and VifIPTrig objects are used for reconstruction underlay network status of app instances later. IPAssignments may not have ip entries for mac corresponding to particular apps, and that is when we create new entry in it's table. Where as VifIPTrig are per mac entries that we always update when a new IP address is found to be associated with it.

Reply:
If VifIPTrig always a superset? 
Reason for asking is that it would be simpler to check and add/update to one thing, and then generate the other thing from that as a separate step.
But that assumes that the first thing has a superset of the info needed for the second thing.

Reply:
I see that due to IPv6+IPv4 the viftrig handling needs to lookup to find the address(es) for the other family as part of updating. That seems to be the only new thing with your code andvifiptrig. Is that correct?

Reply:
Yes correct.